scala-mode-syntax.el

```diff
@@ -155,6 +155,29 @@
           "\\|" scala-syntax:symbolLiteral-re
           "\\|" "null" "\\)"))
 
+(defconst scala-syntax:interpolation-re
+  (concat "\\(" "\\$"  scala-syntax:id-re "\\|" "\\${[^}\n\\\\]*}" "\\)"))
```

Comment:
Yeah, nested interpolation is not going to work.

Unfortunately the parser is not going to work either. It will fail on this case:

    s"foo ${"}"}"

The moral of the story is that syntax highlighting will always be wrong until you write a full-blown shift-reduce parser.